src/battle_message.c

```diff
@@ -4183,116 +4193,116 @@ u32 ShouldDoTrainerSlide(u32 battler, u32 which)
 
     for (i = 0; i < ARRAY_COUNT(sTrainerSlides); i++)
     {
-        if (trainerId == sTrainerSlides[i].trainerId
-            && (((gBattleTypeFlags & BATTLE_TYPE_FRONTIER) && sTrainerSlides[i].isFrontierTrainer)
-                || (!(gBattleTypeFlags & BATTLE_TYPE_FRONTIER) && !sTrainerSlides[i].isFrontierTrainer)))
+        if (trainerId == sTrainerSlides[difficulty]->trainerId
+            && (((gBattleTypeFlags & BATTLE_TYPE_FRONTIER) && sTrainerSlides[difficulty]->isFrontierTrainer)
+                || (!(gBattleTypeFlags & BATTLE_TYPE_FRONTIER) && !sTrainerSlides[difficulty]->isFrontierTrainer)))
```

Comment:
I think this is not correct? Did you want to check `sTrainerSlides[difficulty][i]`? 

Additionally I'd be inclined to encapsulate some of these monstrous statements, e.g. by the following API:

```c
static inline const struct GetTrainerSlide(u32 trainerId);
```
which returns null if it did not find an element? Not 100% sold whether this should be a requirement for your PR.

Reply:
https://github.com/rh-hideout/pokeemerald-expansion/blob/8c3531ffc7049b351011a5ae5be120e1730fd049/src/battle_message.c#L4186

Believe me, I was just as surprised as you. I desperately want to refactor this but I did not want to further bloat this PR. 

It's something I will absolutely commit to fixing soon (and that's a promise because my game heavily relies on this feature).

Reply:
I guess I get it now, I don't really like that this is an array of size TRAINERS_COUNT * DIFFICULTY_COUNT, but maybe that's for another PR to fix. I'm thinking having access to this in trainers.party would be a more user friendly approach. At the very least it could be a linear search again, it was before.